### PR TITLE
Add StateMachineAlias and StateMachineVersion

### PR DESCRIPTION
**Pull request by hongkuntian**
*Issue #, if available:* N/A

*Description of changes:*

Adding the StateMachineAlias and StateMachineVersion resources to the repo.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -16,4 +16,13 @@ repos:
...
+    exclude: |
+      (?x)^(
+      codebuild-ci.yaml|
+      .*dependencies\.yml$
+      )$
+- repo: https://github.com/aws-cloudformation/cfn-lint
+  rev: v0.83.3
+  hooks:
+    - id: cfn-lint
+      files: .*dependencies\.yml$
```
Comment (hongkuntian) [inline .pre-commit-config.yaml:28]: The only notable change in this PR.

The new files introduced Contract Test v2 artifacts for both StateMachineAlias and StateMachineVersion, and we now use a CloudFormation template (yml) to describe the necessary dependencies to run the contract tests.

check-yaml fails on the CloudFormation yml templates with the following:
`could not determine a constructor for the tag '!Ref'`

This change ensures check-yaml excludes `dependencies.yml` files and replaces the check on these files with cfn-lint instead.

**Comment by hongkuntian** [comment]
Forgot to add the new resources in `.github/workflows/pr-ci.yml`

Currently only runs mvn clean verify for Activity and StateMachine:

```
      - name: Run mvn clean verify for Activity and State Machine resources
        run: |
          cd ./activity
          mvn -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -B clean verify
          cd ..
          cd ./statemachine
          mvn -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -B clean verify
```

Fixing this shortly

File: statemachinealias/src/test/java/com/amazonaws/stepfunctions/cloudformation/statemachinealias/ClientBuilderTest.java
Comment (hongkuntian) [inline statemachinealias/src/test/java/com/amazonaws/stepfunctions/cloudformation/statemachinealias/ClientBuilderTest.java:1]: Added `ClientBuilderTest` class to both resources to get branch coverage to satisfy jacoco:check.

We didn't have sufficient branch coverage and were falling below the 80% branch coverage threshold.

File: statemachinealias/pom.xml
Comment (hongkuntian) [inline statemachinealias/pom.xml:1]: Upgraded lombok and jacoco-maven-plugin to get `mvn verify` to run successfully on my local machine.

File: statemachinealias/src/test/java/com/amazonaws/stepfunctions/cloudformation/statemachinealias/ClientBuilderTest.java
Comment (hongkuntian) [inline statemachinealias/src/test/java/com/amazonaws/stepfunctions/cloudformation/statemachinealias/ClientBuilderTest.java:1]: Added `ClientBuilderTest` class to both resources to get branch coverage to satisfy jacoco:check.

We didn't have sufficient branch coverage and were falling below the 80% branch coverage threshold.

File: statemachinealias/pom.xml
Comment (hongkuntian) [inline statemachinealias/pom.xml:1]: Upgraded lombok and jacoco-maven-plugin to get mvn verify to run successfully on my local machine.

**Review by wong-a — APPROVED** [review]

**Review by abdelnn — APPROVED** [review]

**Pull request merged**